src/realm-settings/SessionsTab.tsx

```diff
@@ -0,0 +1,481 @@
+import React, { useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { Controller, useForm, useWatch } from "react-hook-form";
+import {
+  ActionGroup,
+  AlertVariant,
+  Button,
+  FormGroup,
+  PageSection,
+  Switch,
+} from "@patternfly/react-core";
+
+import type RealmRepresentation from "keycloak-admin/lib/defs/realmRepresentation";
+import { FormAccess } from "../components/form-access/FormAccess";
+import { HelpItem } from "../components/help-enabler/HelpItem";
+import { FormPanel } from "../components/scroll-form/FormPanel";
+import { useAdminClient, useFetch } from "../context/auth/AdminClient";
+import { useAlerts } from "../components/alert/Alerts";
+import { useRealm } from "../context/realm-context/RealmContext";
+
+import "./RealmSettingsSection.css";
+import type UserRepresentation from "keycloak-admin/lib/defs/userRepresentation";
+import { TimeSelector } from "../components/time-selector/TimeSelector";
+
+type RealmSettingsSessionsTabProps = {
+  realm?: RealmRepresentation;
+  user?: UserRepresentation;
+};
+
+export const RealmSettingsSessionsTab = ({
+  realm: initialRealm,
+}: RealmSettingsSessionsTabProps) => {
+  const { t } = useTranslation("realm-settings");
+  const adminClient = useAdminClient();
+  const { realm: realmName } = useRealm();
+  const { addAlert } = useAlerts();
+
+  const [realm, setRealm] = useState(initialRealm);
+
+  const {
+    control,
+    handleSubmit,
+    setValue,
+    reset: resetForm,
+    formState,
+  } = useForm<RealmRepresentation>();
+
+  const offlineSessionMaxEnabled = useWatch({
+    control,
+    name: "offlineSessionMaxLifespanEnabled",
+    defaultValue: realm?.offlineSessionMaxLifespanEnabled,
+  });
+
+  useFetch(
+    async () => {
+      const realm = await adminClient.realms.findOne({ realm: realmName });
+
+      return { realm };
+    },
+    (result) => {
+      setRealm(result.realm);
+      setupForm(result.realm);
+    },
+    []
+  );
+
+  useEffect(() => {
+    reset();
+  }, [realm]);
+
+  const setupForm = (realm: RealmRepresentation) => {
+    Object.entries(realm).map((entry) => setValue(entry[0], entry[1]));
```

Comment:
The `entry` can be spread here instead of using array indexes.
```suggestion
    Object.entries(realm).map((entry) => setValue(...entry));
```